Tensile/KernelWriterAssembly.py

```diff
@@ -4160,9 +4223,9 @@ def graShift(self, kernel, tP):
       kStr += inst("s_sub_u32", sgpr(tmpSgpr), sgpr(tmpSgpr), margin, "edge -= margin(%u)"%(margin))
       kStr += inst("v_mov_b32", vgpr(edge), sgpr(tmpSgpr), \
           "edge vgpr = Size%s- WG*MT - margin(%u)"%(tP["tileChar"], margin) )
-      shiftedEdge = self.vgprPool.checkOut(1, "shiftedEdge", self.preventVgprOverflowDuringNewTile)
-      kStr += inst("_v_add_co_u32", vgpr(shiftedEdge), self.vcc, vgpr(edge), self.srdShiftLeft[tc],
-                   "shiftedEdge = edge + srdShiftLeft({})".format(self.srdShiftLeft[tc]))
+      #shiftedEdge = self.vgprPool.checkOut(1, "shiftedEdge", self.preventVgprOverflowDuringNewTile)
+      #kStr += inst("_v_add_co_u32", vgpr(shiftedEdge), self.vcc, vgpr(edge), self.srdShiftLeft[tc],
+      #             "shiftedEdge = edge + srdShiftLeft({})".format(self.srdShiftLeft[tc]))
```

Comment:
Does removing this apply to all cases?

Reply:
Yes. This should work in general cases.
The value of edge is not large and will not go over maximum of int32.
(comment at line 4220. OOB is 0x80000000)
      # edge = (Size - WG*MT) - margin = the last valid load position that won't cause OOB

Then, we can use comparison with int32 type (use v_min_i32 instruction) instead of converting it to unit32 (= ShiftedEdge ).

After I changed comparison with ShiftedEdge to vmin, we no longer need shiftedEdge value in self.groOffsetInMacroTile case.